src/coreComponents/constitutive/permeability/ConstantPermeability.cpp

```
@@ -54,9 +82,8 @@ void ConstantPermeability::allocateConstitutiveData( dataRepository::Group & par
   {
     for( localIndex q = 0; q < numQuad; ++q )
     {
-      m_permeability[ei][q][0] =  m_permeabilityComponents[0];
-      m_permeability[ei][q][1] =  m_permeabilityComponents[1];
-      m_permeability[ei][q][2] =  m_permeabilityComponents[2];
+      for ( int c=0; c < 6; c++)
+      m_permeability[ei][q][c] =  m_symmetricFullPermeabilityTensor[c];
```

Shouldn't we enforce positiveness of diagonal component here ?

---

We should really throw an error earlier if someone inputs a negative value but, for now, I am using the default negative value to distinguish whether a diagonal or a full tensor was provided so I need to improve that logic fist.

---

One option would be to simply provide 2 R1Tensors one for the main diagonal and one for the off diagonal terms and have the off diagonal components default to 0.0 if not provided.

---

Just noticed that the order would be imposed by `LvArray::Ri_eq_symAijBj` implicitly :/

---

yes, and it's different from the one used by `vtk`.